pkgs/applications/version-management/anu/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, fetchCrate
+, rustPlatform
+, pkg-config
+, clang
+, libclang
+, libsodium
+, openssl
+, xxHash
+, zstd
+, CoreServices
+, Security
```

Comment:
These are darwin-specific; might be best to leave them as optional? (Not really sure how this is normally handled, first time I have seen this platform specific dependencies)

```suggestion
, CoreServices ? null
, Security ? null
```

Reply:
Isn't `stdenv.lib.optionals stdenv.isDarwin [ CoreServices Security ])` sufficient? There are some similar instances in nixpkgs. I'd like to hear from experts whether this is idiomatic or not too, though.

Reply:
I tried building it on NixOS and had an issue of insufficient arguments. I also don't know how it is normally handled. :C

Reply:
Ah, I see. Thank you for sharing your situation.

Reply:
Strange, I built it on NixOS just fine without that...

Reply:
20.09?

Reply:
As reference, I am on NixOS 20.09 with `nix-unstable` as the user channel. I used `nix-build -E "with import <nixpkgs> {}; callPackage ./anu.nix { libclang = llvmPackages.libclang; }"` to build this expression and that required the suggested changes.

Reply:
> I'd like to hear from experts whether this is idiomatic or not too, though.

I don't think this is idiomatic, but I'm definitely not an expert.

Reply:
Indeed, passing dependencies in regardless of the platform is the idiomatic approach. We don't reference platform specific dependencies with pkgs.hello instead of passing them in as arguments either.